p4-16/spec/P4-16-spec.mdk

```diff
@@ -5523,6 +5524,37 @@ use to implement generalized switch statements.
 
 [GeneralizedSwitchStatements]: <https://github.com/p4lang/p4-spec/blob/master/p4-16/spec/docs/implementing-generalized-switch-statements.md>
 
+## Loop statement { #sec-loop-stmt }
+
+A `loop` statement executes a statement zero or more times, based on a condition
+or a range or collection.
+
+~ Begin P4Grammar
+
+[INCLUDE=grammar.mdk:loopStatement]
+
+[INCLUDE=grammar.mdk:forInitStatement]
+
+[INCLUDE=grammar.mdk:forUpdateStatement]
+
+~ End P4Grammar
+
+The basic 3-clause `for` statement is similar to a C for statement.  The init statements
+will be executed prior to executing the loop.  The condition will be evaluated before
+the first iteration and after each iteration, to determine if the loop should exit.
+The update statements will be executed after the loop body and before evaluating the
+condition again for the next iteration.
+
+The `for`-`in` statement executes the body once for each value in a range or each
+element in a list expression or header stack.
+
+A `break;` statement may be executed in a loop body to immediately exit the loop,
+without executing the rest of the body or the update statements.
+
+A `continue;` statement may be executed in a loop body to skip the rest of the current
+loop body, executing the update statements and then evaluating the condition to
+either execute the loop body again, or terminate the loop.
+
```

Comment:
Do you think it is worth explicitly saying that any loop variable preceded by a type has a scope that is limited to the loop statement?

Reply:
Perhaps just say that the scope of anything declared within a loop statement is limited to the loop statement?  That is consistent with (recent) C++ and logical if we ever add C++ style if statements with initializers.